src/Scripting/Core/Hosting/AssemblyLoader/InteractiveAssemblyLoader.cs

```diff
@@ -91,11 +90,23 @@ public InteractiveAssemblyLoader(MetadataShadowCopyProvider? shadowCopyProvider
             _loadedAssembliesBySimpleName = new Dictionary<string, List<LoadedAssemblyInfo>>(AssemblyIdentityComparer.SimpleNameComparer);
             _dependenciesWithLocationBySimpleName = new Dictionary<string, List<AssemblyIdentityAndLocation>>();
 
-            _runtimeAssemblyLoader = AssemblyLoaderImpl.Create(this);
+#if NET
+            _runtimeAssemblyLoader = new CoreAssemblyLoaderImpl(this, isCollectible);
+#else
+            if (isCollectible)
+            {
+                throw new InteractiveAssemblyLoaderException(ScriptingResources.CollectibleAssembliesNotSupported);
+            }
+
+            _runtimeAssemblyLoader = new DesktopAssemblyLoaderImpl(this);
+#endif
         }
 
         public void Dispose()
         {
+            // This field reference loaded assemblies, so it must be cleared before the loader is disposed.
+            // Otherwise, AssemblyLoadContext.Unload in CoreAssemblyLoaderImpl would not working
+            _loadedAssembliesBySimpleName.Clear();
```

Comment:
I didn't find any this Dispose's invoke places in roslyn repo, I don't sure clear it's safe or suitable to clear  `_loadedAssembliesBySimpleName` , But in my test case i don't see any negative behavior